test/jpypetest/test_startup.py

```diff
@@ -90,14 +98,26 @@ def testClasspathBadType(self):
         with self.assertRaises(TypeError):
             runStartJVMTest(classpath=1, convertStrings=False)
 
-    def testPathArg(self):
+    def testJVMPathArg_Str(self):
         runStartJVMTest(self.jvmpath, classpath=cp, convertStrings=False)
 
-    def testPathKeyword(self):
-        path = jpype.getDefaultJVMPath()
+    def testJVMPathArg_Path(self):
+        with self.assertRaises(TypeError):
+            runStartJVMTest([
+                # Pass a path as the first argument. This isn't supported (this is
+                # reflected in the type definition), but the fact that it "works"
+                # gives rise to this test.
+                Path(self.jvmpath), cp],  # type: ignore
```

Comment:
why shouldn't we just allow lists as well? or sequences of path like types

Reply:
The first element of the variable arguments is the ``jvmpath``. Looking at the code, I got the impression that rather than relying on positional arguments to pass this information (as was the case pre v0.7), it is preferred to use the named keyword arguments. So you would do ``startJVM(jvmpath=path_or_str, ...)``. Same for the ``classpath`` keyword argument.

In essence, I understood that the variable arguments handling is the legacy interface, and the keyword arguments are the preferred approach. I therefore provided better typing information for the keyword arguments.

Technically, whilst a little bit more difficult to do, I think it would be possible to add this *new* behaviour to the variable arguments form, if you think it is important?